driver.go

```diff
@@ -1,81 +1,142 @@
 package server
 
 import (
-	"bufio"
-	"io"
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"syscall"
+	"time"
 
 	"github.com/bblfsh/server/runtime"
-
-	"github.com/Sirupsen/logrus"
+	"github.com/opencontainers/runc/libcontainer/configs"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
 	"gopkg.in/bblfsh/sdk.v1/protocol"
-	"gopkg.in/bblfsh/sdk.v1/protocol/driver"
 )
 
-// Driver is a client to communicate with a driver. It provides the parser
-// interface and is closeable.
 type Driver interface {
-	protocol.Parser
-	io.Closer
+	Start() error
+	Stop() error
+	Service() protocol.ProtocolServiceClient
+}
+
+type DriverInstance struct {
+	ctx  context.Context
+	p    *runtime.Process
+	c    runtime.Container
+	i    runtime.DriverImage
```

Comment:
`proc` and `img` may be better names than `p` and `i`.

Reply:
ok